source/fe/mapping_q_cache.cc

```diff
@@ -125,6 +129,52 @@ MappingQCache<dim, spacedim>::initialize(
 
 
 
+template <int dim, int spacedim>
+void
+MappingQCache<dim, spacedim>::initialize(
+  const Triangulation<dim, spacedim> &tria,
+  const std::function<Point<spacedim>(const Point<spacedim> &)>
+    &transformation_function)
+{
+  MappingQ1<dim, spacedim>  mapping_base;
+  FE_Nothing<dim, spacedim> fe;
+  QGaussLobatto<dim>        quadrature_gl(this->polynomial_degree + 1);
+
+  std::vector<Point<dim>> quadrature_points;
+  for (const auto i : FETools::hierarchic_to_lexicographic_numbering<dim>(
+         this->polynomial_degree))
+    quadrature_points.push_back(quadrature_gl.point(i));
+  Quadrature<dim> quadrature(quadrature_points);
+
+  Threads::ThreadLocalStorage<std::unique_ptr<FEValues<dim, spacedim>>>
+    fe_values_all;
+
+  this->initialize(
+    tria,
+    [&](const typename Triangulation<dim, spacedim>::cell_iterator &cell) {
+      auto &fe_values = [&]() -> FEValues<dim, spacedim> & {
+        auto &fe_values = fe_values_all.get();
+        if (fe_values.get() == nullptr)
+          fe_values = std::make_unique<FEValues<dim, spacedim>>(
+            mapping_base, fe, quadrature, update_quadrature_points);
+
+        return *fe_values;
+      }();
+
+      fe_values.reinit(cell);
+      auto points = fe_values.get_quadrature_points();
```

Comment:
For the case we prefer to use the points of `MappingQGeneric`, I believe we could simplify this quite a bit (I did not check if it runs well or if I overlooked something here):
```suggestion
  this->initialize(
    tria,
    [&](const typename Triangulation<dim, spacedim>::cell_iterator &cell) {
      auto points = this->MappingQGeneric<dim>::compute_mapping_support_points(cell);
```